addons/predicates/addon_config_test.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package predicates
+
+import (
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/constants"
+	cniv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/cni/v1alpha1"
+)
+
+var _ = Describe("Addon config annotation predicate", func() {
```

Comment:
**praise**: Like this predicate test being added apart from env test. We should backfill some more predicates test in the future for other predicates